src/f3dasm/_src/datageneration/datagenerator.py

```diff
@@ -43,15 +44,32 @@ class DataGenerator:
     """Base class for a data generator"""
```

Comment:
Interface should have a complete docstring explaining the sequence and expected functionality of the different methods involved in ._run().

It must be clear how the ExperimentSample object is stored and should be handled on each of these methods.

Reply:
I've added more information about how the run function chains these methods together